addons/pos_event/models/event_registration.py

```diff
@@ -8,6 +9,26 @@ class EventRegistration(models.Model):
     pos_order_id = fields.Many2one(related='pos_order_line_id.order_id', string='PoS Order')
     pos_order_line_id = fields.Many2one('pos.order.line', string='PoS Order Line', ondelete='cascade', copy=False)
 
+    @api.depends('pos_order_id.state', 'pos_order_id.currency_id', 'pos_order_id.amount_total')
+    def _compute_registration_status(self):
+        if not self.pos_order_id:
+            super()._compute_registration_status()
+            return
+
+        for pos_order, registrations in self.grouped('pos_order_id').items():
+            cancelled_so_registrations = registrations.filtered(lambda reg: reg.pos_order_id.state == 'cancel')
+            cancelled_so_registrations.state = 'cancel'
+            cancelled_registrations = cancelled_so_registrations | registrations.filtered(lambda reg: reg.state == 'cancel')
+            if not pos_order or float_is_zero(pos_order.amount_total, precision_rounding=pos_order.currency_id.rounding):
+                registrations.sale_status = 'free'
+                registrations.filtered(lambda reg: not reg.state or reg.state == 'draft').state = "open"
```

Comment:
Didn't mean to submit quite yet so loose comment here.

Would maybe let this be the default for the compute when `event_product` is installed, and not do anything in this case. To avoid overriding the default of the other compute, and conversly.

Reply:
On it